File: app/src/main/java/com/elishaazaria/sayboard/settingsfragments/LogicSettingsFragment.java
package com.elishaazaria.sayboard.settingsfragments;

import android.os.Bundle;

import androidx.preference.PreferenceFragmentCompat;
import androidx.preference.PreferenceManager;

import com.elishaazaria.sayboard.AppCtx;
import com.elishaazaria.sayboard.R;

public class LogicSettingsFragment extends PreferenceFragmentCompat {

    @Override
    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
        PreferenceManager preferenceManager = getPreferenceManager();
        preferenceManager.setSharedPreferencesName(AppCtx.getStringRes(R.string.main_shared_pref));
        setPreferencesFromResource(R.xml.logic_preferences, rootKey);
    }
}

File: app/src/main/java/com/elishaazaria/sayboard/settingsfragments/LogicSettingsFragment.kt
package com.elishaazaria.sayboard.settingsfragments

import android.os.Bundle
import androidx.preference.PreferenceFragmentCompat
import com.elishaazaria.sayboard.AppCtx.getStringRes
import com.elishaazaria.sayboard.R

class LogicSettingsFragment : PreferenceFragmentCompat() {
    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
        val preferenceManager = preferenceManager
        preferenceManager.sharedPreferencesName = getStringRes(R.string.main_shared_pref)
        setPreferencesFromResource(R.xml.logic_preferences, rootKey)
    }
}